python/hail/expr/aggregators.py

```diff
@@ -129,13 +566,91 @@ def hardy_weinberg(expr):
 
 @typecheck(expr=oneof(expr_list, expr_set))
 def explode(expr):
+    """Explode an array or set expression to aggregate the elements of all records.
+
+    Examples
+    --------
+    Compute the mean of all elements in fields `C1`, `C2`, and `C3`:
+
+    .. doctest::
+
+        >>> table1.aggregate(mean_c = agg.mean(agg.explode([table1.C1, table1.C2, table1.C3])))
+        Struct(mean_c=24.8333333333)
+
+    Compute the set of all observed elements in the `filters` field (``Set[String]``):
+
+    .. doctest::
+
+        >>> dataset.aggregate_rows(filters = agg.collect_as_set(agg.explode(dataset.filters)))
+        Struct(filters=set([u'VQSRTrancheSNP99.80to99.90',
+                            u'VQSRTrancheINDEL99.95to100.00',
+                            u'VQSRTrancheINDEL99.00to99.50',
+                            u'VQSRTrancheINDEL97.00to99.00',
+                            u'VQSRTrancheSNP99.95to100.00',
+                            u'VQSRTrancheSNP99.60to99.80',
+                            u'VQSRTrancheINDEL99.50to99.90',
+                            u'VQSRTrancheSNP99.90to99.95',
+                            u'VQSRTrancheINDEL96.00to97.00']))
+
+    Notes
+    -----
+    This method can be used with aggregator functions to aggregate the elements
+    of collection types (:class:`hail.expr.types.TArray` and :class:`hail.expr.types.TSet`).
+
+    The result of the :meth:`explode` and :meth:`filter` methods is an
+    :class:`Aggregable` expression which can be used only in aggregator
+    methods.
+
+    Parameters
+    ----------
+    expr : :class:`hail.expr.expression.CollectionExpression`
+        Expression of type :class:`hail.expr.types.TArray` or :class:`hail.expr.types.TSet`.
+
+    Returns
+    -------
+    :class:`Aggregable`
+        Aggregable expression.
```

Comment:
`Aggregable` not linked in docs... needs moar docstring?

Reply:
will add this in another PR. Not quite sure what to say yet.